packages/react-reconciler/src/ReactFiberHooks.old.js

```diff
@@ -721,6 +735,91 @@ function createFunctionComponentUpdateQueue(): FunctionComponentUpdateQueue {
   };
 }
 
+function use<T>(usable: Usable<T>): T {
+  if (
+    usable !== null &&
+    typeof usable === 'object' &&
+    typeof usable.then === 'function'
+  ) {
+    // This is a thenable.
+    const thenable: Thenable<T> = (usable: any);
+
+    // Track the position of the thenable within this fiber.
+    const index = thenableIndexCounter;
+    thenableIndexCounter += 1;
+
+    switch (thenable.status) {
+      case 'fulfilled': {
+        const fulfilledValue: T = (thenable.value: any);
+        return fulfilledValue;
+      }
+      case 'rejected': {
+        const rejectedError: mixed = thenable.value;
```

Comment:
This public API is intentional? Promise.allSettled calls it `reason` not `value` for rejections.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
Yes. The publicness makes it so that we (e.g. Flight) and other libraries can produce already resolved Promises that can be fast. Controversial as it may be. It's time this existed.

Reply:
Probably should use `reason` as the prop then.

Reply:
Yeah current plan is to make this a public API, but these aren't final. We did pick "fulfilled" and "rejected" based on `allSettled`, though. Having separate fields seems fine, a bit annoying that it's two fields instead of one but I suppose it makes the types easier to model anyway.

Reply:
Switched it to use `reason`